### PR TITLE
Add support of Job Names with folders (issue #210) + Add JobDSL plugin to the Vanilla Docker image

### PR DESCRIPTION
**Pull request by addihorn**
allows parameter ```--job-name``` to create a job inside a folder-structure.
The parameter will be splitted at ```/``` to allow creation and assignment of jobs to specific folders.

Folders can be created inside ```Runner``` (if not created otherwise) or by Job DSL to allow introduction of additional library-settings etc.

### CODE REVIEW

**Review by oleg-nenashev — COMMENTED** [review]
I am fine with adding Johb DSL to the vanilla package, as long as it is not a mandatory dependency. Actually it is even better since we get support for job definitions in JCasC. Maybe it is another mode to be added later: triggering job from JCasC or Groovy hooks instead of passing Jenkinsfile.  I will create a follow-up ticket, no need to do it here unless you want to do it shortly.

@addihorn before merging, could you please align the codestyle with the existing codebase?

E.g. 

```java
if (condition) {
 // ...
}
```

**Comment by addihorn** [comment]
hi @oleg-nenashev 

Thanks for the review.
Just to understand you correctly
>I am fine with adding Johb DSL to the vanilla package, as long as it is not a mandatory dependency.

You want me, to mark the JobDSL as optional dependency in the vanilla package pom, or is it okay, since the plugin is not referenced in payloads?

**Comment by oleg-nenashev** [comment]
Sorry for confusion. The current version looks good to me function-wise. Just needs some codestyle cleanup. Happy to merge/release after that

**Comment by addihorn** [comment]
Thank you for the clarification. Just wanted to make sure.

Alligned the code with your existing code base, as you asked.

File: demo/job-folders/README.md
Comment (oleg-nenashev) [inline demo/job-folders/README.md]:
```suggestion
	jenkins4eval/jenkinsfile-runner:dev \
```

typo

File: payload/src/main/java/io/jenkins/jenkinsfile/runner/Runner.java
Comment (oleg-nenashev) [inline payload/src/main/java/io/jenkins/jenkinsfile/runner/Runner.java:139]: Would be nice to add logging there. This is a wrong state which needs to be reported, and it should lead to the execution failure IMHO

File: demo/job-folders/JenkinsfileWithLibraryCall
Comment (oleg-nenashev) [inline demo/job-folders/JenkinsfileWithLibraryCall]:
```suggestion
```

**Review by oleg-nenashev — COMMENTED** [review]
Would be nice to also address the error propagation before the merge

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me. Thanks for addressing the comments @addihorn !

**Pull request merged**
